tests/unit/ai/test_cursor_provider.py

```
@@ -0,0 +1,259 @@
+        """Raise AIAuthenticationError on auth failure stderr."""
+        with (
+            patch("subprocess.run") as mock_run,
+            pytest.raises(AIAuthenticationError, match="login"),
+        ):
+            mock_run.return_value = subprocess.CompletedProcess(
+                args=[],
+                returncode=1,
+                stdout="",
+                stderr="Authentication required. Run 'agent login' first.",
+            )
+            provider.complete("Hello")
```

Ruff (0.15.20)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)